sklearn/tree/_tree.pyx

```diff
@@ -652,8 +652,15 @@ cdef class Tree:
 
         value_shape = (node_ndarray.shape[0], self.n_outputs,
                        self.max_n_classes)
+
+        if (node_ndarray.dtype != NODE_DTYPE):
+            swapped_node_ndarray = node_ndarray.byteswap().newbyteorder();
```

Comment:
```suggestion
        if (node_ndarray.dtype != NODE_DTYPE):
            # possible mismatch of big/little endian due to serialization
            # on a different architecture. Try swapping the byte order.  
            swapped_node_ndarray = node_ndarray.byteswap().newbyteorder();
```

Reply:
Also this could be simplified to
```py
        if (node_ndarray.dtype != NODE_DTYPE):
            # <comment from above>
            node_ndarray = node_ndarray.byteswap().newbyteorder();
            if (node_ndarray.dypte != NODE_DTYPE):
                raise ...
```

Reply:
Thanks @rth! Code simplified and committed.